src/robotFactory.js

```diff
@@ -1,17 +1,89 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
-}
+BaseRobot.prototype = {
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  },
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  },
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  },
 
-function DeliveryDrone() {
-  // implement
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  },
+
+  getInfo() {
+    return `Robot: ${this.name},`
+    + ` Chip version: ${this.chipVersion},`
+    + ` Weight: ${this.weight}`;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+
+  this.coords.z = 0;
 }
 
+FlyingRobot.prototype = {
+  goUp(step = 1) {
+    this.coords.z += step;
+
+    return this;
+  },
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+
+    return this;
+  },
+};
+
+function DeliveryDrone(name, weight, coords, chipVersion,
```

Comment:
```suggestion
function DeliveryDrone(
  name,
  weight,
  coords,
  chipVersion,
```
Якщо переносиш параметри, то перенось всі. Так зручніше читається